src/microbe_stage/editor/EditorActionHistory.cs

```diff
@@ -0,0 +1,110 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+public class EditorActionHistory : ActionHistory<MicrobeEditorAction>
+{
+    private List<MicrobeEditorActionData> cache;
+
+    public int WhatWouldActionCost(MicrobeEditorActionData action)
+    {
+        int actionCost = action.CalculateCost();
+        int result = actionCost;
+        foreach (var historyAction in cache ??= GetActionHistorySinceLastNewMicrobePress())
+        {
+            switch (action.GetInterferenceModeWith(historyAction))
+            {
+                case MicrobeActionInterferenceMode.Combinable:
+                    result -= actionCost;
+                    result -= historyAction.CalculateCost();
+                    result += action.Combine(historyAction).CalculateCost();
+                    break;
+                case MicrobeActionInterferenceMode.CancelsOut:
+                    result -= actionCost;
+                    result -= historyAction.CalculateCost();
+                    return result;
+                case MicrobeActionInterferenceMode.ReplacesOther:
+                    result -= historyAction.CalculateCost();
+                    break;
+            }
+        }
+
+        return result;
+    }
+
+    /// <summary>
+    ///   Calculates the remaining MP from the action history
+    /// </summary>
+    /// <returns>The remaining MP</returns>
+    public int CalculateMutationPointsLeft()
+    {
+        var copyLength = (cache ??= GetActionHistorySinceLastNewMicrobePress()).Count;
+        for (int compareToIndex = 0; compareToIndex < copyLength - 1; compareToIndex++)
+        {
+            for (int compareIndex = compareToIndex + 1; compareIndex < copyLength; compareIndex++)
+            {
+                switch (cache[compareIndex].GetInterferenceModeWith(cache[compareToIndex]))
+                {
+                    case MicrobeActionInterferenceMode.NoInterference:
+                        break;
+                    case MicrobeActionInterferenceMode.Combinable:
+                        var combinedValue = cache[compareIndex].Combine(cache[compareToIndex]);
+                        cache.RemoveAt(compareIndex);
+                        cache.RemoveAt(compareToIndex);
+                        compareIndex--;
```

Comment:
compareIndex -= 2 ?

Reply:
nope, check one line below. This also reduces the `compareIndex` after the `Insert` call